src/dmd/func.d

```diff
@@ -1511,11 +1521,19 @@ extern (C++) class FuncDeclaration : Declaration
                 /* Drill down and check the struct's fields
                  */
                 auto sym = t.toDsymbol(null).isStructDeclaration();
+                const tName = t.toChars.toDString;
+                const entry = parentTypes.insert(tName, t);
+                if (entry == null)
```

Comment:
Why is this using a `StringTable` on `toChars` instead of a `DsymbolTable`? The generated string is not always unique, so the check will give false positives.

Reply:
I don't think I considered a symbol table.  Might be worth looking into now

Reply:
I'm trying using a `DsymbolTable` now for `traverseIndirections`, see https://github.com/dlang/dmd/pull/12885